src/services/scimTransformation.ts

```diff
@@ -0,0 +1,240 @@
+import { APIGatewayProxyEventHeaders } from 'aws-lambda';
+import { getTenancyAndGroupFromPath } from '../common/utils';
+import {
+  ScimPatchOperation,
+  ScimPutOperation,
+  ScimGroupOperations,
+} from '../types/scim';
+import {
+  fetchUsersInGroup,
+  fetchAllUsers,
+  fetchUserGroupRelationship,
+  AllowMethods,
+} from './scimFetch';
+
+export type MetaPayload<T = unknown> = {
+  data: T;
+  headers: APIGatewayProxyEventHeaders;
+  method: AllowMethods;
+  path: string;
+};
+
+/**
+ * Replaces the ForgeRock member operation replace with an AWS certified
+ * operation.
+ *
+ * @param membersList forgerock members list
+ * @param groupPath location of resource
+ * @param headers apigateway headers for proxy
+ * @returns AWS accepted operations for members update
+ */
+export const convertMemberReplaceToSeparateOps = async (
+  membersList: ReadonlyArray<string>,
+  groupPath: string,
+  headers: APIGatewayProxyEventHeaders
+): Promise<
+  ReadonlyArray<ScimPatchOperation['Operations'][number]> | undefined
+> => {
+  const [rawPath, groupId] = getTenancyAndGroupFromPath(groupPath) ?? [
+    undefined,
+  ];
+
+  if (rawPath === undefined || groupId === undefined) {
+    return undefined;
+  }
+
+  const currentGroupMembers = await fetchUsersInGroup(
```

Comment:
In `scimTransformation` can you pass this in as a function instead?

Reply:
Move all fetch function definitions out of the transformation logic back up to the handler